### PR TITLE
OHRM5X-1093: Employee Attendance Record API Delete Endpoint

### PR DESCRIPTION
**Pull request by SandeepaOHRM**
This PR contains:-
Refactored Employee Attendance Record API delete endpoint
Improved Test coverage for delete endpoint

### CODE REVIEW

File: symfony/plugins/orangehrmAttendancePlugin/Api/EmployeeAttendanceRecordAPI.php [hunk 1]
```diff
@@ -250,38 +250,41 @@ protected function getCommonValidationRules(): array
...
+            if (empty($attendanceRecordIds)) {
+                throw AttendanceServiceException::deletableAttendanceRecordIdsEmpty();
+            }
+            $userAllowedAttendanceRecords = $this->getAttendanceService()
+                ->getAttendanceDao()
+                ->getAttendanceRecordsByEmpNumberAndIds($attendanceRecordOwnedEmpNumber, $attendanceRecordIds);
+            $userAllowedAttendanceRecordIds = [];
+            foreach ($userAllowedAttendanceRecords as $userAllowedAttendanceRecord) {
+                $userAllowedAttendanceRecordIds[] = $userAllowedAttendanceRecord->getId();
+            }
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmAttendancePlugin/Api/EmployeeAttendanceRecordAPI.php:275]: :relaxed: 
```suggestion
            $userAllowedAttendanceRecordIds = array_map(fn (AttendanceRecord $attendanceRecord) => $attendanceRecord->getId(), $userAllowedAttendanceRecords);
```

File: symfony/plugins/orangehrmAttendancePlugin/Exception/AttendanceServiceException.php [hunk 2]
```diff
@@ -78,4 +78,12 @@ public static function punchOutDateTimeNull(): self
     {
         return new self('Punch Out Date And Time Should Not Be Null');
     }
+
+    /**
+     * @return static
+     */
+    public static function deletableAttendanceRecordIdsEmpty(): self
+    {
+        return new self('No IDS Found');
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmAttendancePlugin/Exception/AttendanceServiceException.php:87]:
```suggestion
        return new self('No IDs Found');
```

File: symfony/plugins/orangehrmAttendancePlugin/Api/EmployeeAttendanceRecordAPI.php [hunk 3]
```diff
@@ -250,38 +250,41 @@ protected function getCommonValidationRules(): array
...
+            return new EndpointResourceResult(ArrayModel::class, $attendanceRecordIds);
+        } catch (AttendanceServiceException $e) {
+            throw $this->getBadRequestException($e->getMessage());
         }
-        $this->getAttendanceService()->getAttendanceDao()->deleteAttendanceRecords($attendanceRecordIds);
-        return new EndpointResourceResult(ArrayModel::class, $attendanceRecordIds);
     }
 
+
+
```
Comment (RajithaKumara) [inline symfony/plugins/orangehrmAttendancePlugin/Api/EmployeeAttendanceRecordAPI.php:287]: ?

**Review by RajithaKumara — APPROVED** [review]

File: symfony/plugins/orangehrmAttendancePlugin/Api/EmployeeAttendanceRecordAPI.php (hunk 1 above)
Comment (SandeepaOHRM) [reply symfony/plugins/orangehrmAttendancePlugin/Api/EmployeeAttendanceRecordAPI.php:275]: New thing, done

File: symfony/plugins/orangehrmAttendancePlugin/Exception/AttendanceServiceException.php (hunk 2 above)
Comment (SandeepaOHRM) [reply symfony/plugins/orangehrmAttendancePlugin/Exception/AttendanceServiceException.php:87]: done

**Review by RajithaKumara — APPROVED** [review]

**Pull request merged**